apps/cowswap-frontend/src/modules/hooksStore/dapps/AirdropHookApp/hooks/usePreviewClaimableTokens.ts

```diff
@@ -0,0 +1,76 @@
+import { current } from "@reduxjs/toolkit";
+import { useCallback } from "react"
+
+interface PreviewClaimableTokensParams {
+    link:string,
+    address:string,
+}
+
+function findIntervalKey(name:string, intervals:object) {
+    /* function to check if a name is inside a interval
+    intervals is in the format: {
+        "name1":"name2",
+        "name3":"name4",
+        ...
+    }
+    name4 > name3 > name2 > name1
+
+    returns the interval key if the condition is checked, else undefined
+    */
+    const keys = Object.keys(intervals);
+    const numberOfKeys = keys.length
+    let currentKeyIndex = 0
+    let found = false
+    while (!found) {
+      const currentKey = keys[currentKeyIndex];
+      if ((currentKey <= name) && (intervals[currentKey] >= name)) {
+        return currentKey
+        found = true
+      }
+      if (currentKey > name) {
+        return undefined
+      }
+      currentKeyIndex += 1
+      if (currentKeyIndex > numberOfKeys) {
+        return undefined;
+      }
+    }
+    return undefined;
+  }
```

Comment:
It would be nice to have some tests for this function

Reply:
I don't know how to create formal tests in the cowswap environment, but I added a draft